sematic/ui/src/types/Types.tsx

```diff
@@ -541,14 +545,81 @@ function PlotlyFigureValueView(props: ValueViewProps) {
 
 function MatplotlibFigureValueView(props: ValueViewProps) {
   let { valueSummary } = props;
-  let { path } = valueSummary;
+
+  const hasFigureJsonData = useMemo(() => !!valueSummary['mpld3'], [valueSummary]);
+  
+  return hasFigureJsonData ? 
+    <MatplotlibFigureValueFigure key={valueSummary['mpld3']['id']} spec={valueSummary['mpld3']} />
+    : <MatplotlibFigureValueImage path={valueSummary['path']} /> ;
+}
+
+interface MatplotlibFigureValueImageProps{
+  path: string
+}
+
+function MatplotlibFigureValueImage(props: MatplotlibFigureValueImageProps) {
+  const { path } = props;
   return (
     <Zoom>
       <img src={path} width={"100%"} alt="matplotlib figure" />
     </Zoom>
   );
 }
 
+interface MatplotlibFigureValueFigureProps{
+  spec: any
+}
+
+function MatplotlibFigureValueFigure(props: MatplotlibFigureValueFigureProps) {
+  const { spec: specProp } = props;
+  const figureId = useMemo(() => `fig-${uuidv4()}`, []);
+
+  const drawFigure = useMatplotLib(figureId);
+
+  const [figDivRef, { width, height}] = useMeasure<HTMLDivElement>();
+
+  const hasFigureRendered = useRef(false)
+
+  const [scaleAndTranslate, setScaleAndTranslate] = useState({
+    scale: 1, translate: {x: 0, y: 0}
+  });
+
+  const [scaledHeight, setScaleHeight] = useState(0);
+
+  useLayoutEffect(() => {
+    // wait for the element's final layout when its width is fully expanded.
+    if (!hasFigureRendered.current && width > 0) {
+      const spec = {
+        ...specProp,
+        plugins: []
+      };
+
+      const scaledHeight = width / spec.width * spec.height;
+      
+      drawFigure(spec);
+
+      const scale = width / spec.width;
+
+      setScaleHeight(scaledHeight);
+      setScaleAndTranslate({
```

Comment:
The server side could not foresee what the FE would require, and rendered a super large plot (in logical size conveyed by the JSON format). Therefore, if we don't manually shrink it down by actually measuring the size of the display area, it is gonna look very bad. Here is a screenshot of how bad:

<img width="782" alt="image" src="https://user-images.githubusercontent.com/1046489/203588430-3e4fda3b-de7a-47c3-8372-5bfcc369fca8.png">